File: Lab07/lab07.js
function sendRequest() {
    var ans = "";
    var url = "../../cgi-bin/lab07.py";
    if (document.getElementById("winter").checked == true)
        ans = "winter";
    else if (document.getElementById("spring").checked == true)
        ans = "spring";
    else if (document.getElementById("summer").checked == true)
        ans = "summer";
    else if (document.getElementById("fall").checked == true)
        ans = "fall";
    else
        document.getElementById("alertDiv").innerHTML = "Nie wybrano opcji!";

    if (ans != "") {
        document.getElementById("alertDiv").innerHTML = "";

		const xhr = new XMLHttpRequest();
        var data = encodeURI("ans=" + ans);
		var graph = document.getElementById("graphDiv");
		url += "?ans=" + ans;
		
		xhr.open("GET", url, true);
 	  	xhr.addEventListener("load", e => {
    	if ( xhr.status == 200 )  {
				graph.innerHTML = xhr.responseText;
    		}
  		})
		
        xhr.send(data);
		document.getElementById(ans).checked = false;
    }
}

File: Lab07/lab07.py
#!/usr/bin/python3

import cgi
form = cgi.FieldStorage()
data = form.getvalue("ans", "---")
content = []

with open('../Labs/Lab07/data_graph.txt') as f:
    line = f.read()
    numbers = line.split()
    for x in numbers:
        content.append(int(x))

if(data == "winter"):
    content[0] += 1
elif(data == "spring"):
    content[1] += 1
elif(data == "summer"):
    content[2] += 1
elif(data == "fall"):
    content[3] += 1

content_max = max(content)

f = open('../Labs/Lab07/data_graph.txt', 'w')
for x in content:
    f.write(str(x) + " ")
f.close()

print ("Content-Type: text/html")
print ()
print("<svg>")
print("<rect id=\"1\" fill=\"black\" height=\"")
print(200*content[0]/content_max)
print("\" y=\"")
print(200 - 200*content[0]/content_max)
print("\"/>")
print("</svg>")
print("<svg>")
print("<rect id=\"2\" fill=\"green\" height=\"")
print(200*content[1]/content_max)
print("\" y=\"")
print(200 - 200*content[1]/content_max)
print("\"/>")
print("</svg>")
print("<svg>")
print("<rect id=\"3\" fill=\"yellow\" height=\"")
print(200*content[2]/content_max)
print("\" y=\"")
print(200 - 200*content[2]/content_max)
print("\"/>")
print("</svg>")
print("<svg>")
print("<rect id=\"4\" fill=\"brown\" height=\"")
print(200*content[3]/content_max)
print("\" y=\"")
print(200 - 200*content[3]/content_max)
print("\"/>")
print("</svg>")
print("<div id=\"titlesDiv\">")
print("<p class=\"small\">Zima</p>")
print("<p class=\"small\">Wiosna</p>")
print("<p class=\"small\">Lato</p>")
print("<p class=\"small\">Jesien</p>")
print("</div>")

